CONTRIBUTING.md

```diff
@@ -0,0 +1,176 @@
+
+1. Install ``docker``, launch it.
+
+2. Clone the [ansible-core](https://github.com/ansible/ansible) repository:
+```bash
+git clone https://github.com/ansible/ansible.git
+```
+3. Go to the cloned repository and prepare the environment:
+```bash
+cd ansible && source hacking/env-setup
+pip install -r requirements.txt
+cd ~
+```
+4. Create the following directories in your home directory:
+```bash
+mkdir -p ~/ansible_collections/community/mysql
+```
+
+5. Fork the ``community.mysql`` repository via the GitHub web interface.
+
+6. Clone the forked repository from your profile to the created path:
+```bash
+git clone https://github.com/YOURACC/community.mysql.git ~/ansible_collections/community/mysql
+```
+
+7. Go there:
+```bash
+cd ~/ansible_collections/community/mysql
+```
+
+8. Be sure you are in the main branch:
+```bash
+git status
+```
+
+9. Show remotes. There should be the ``origin`` repository only:
+```bash
+git remote -v
+```
+
+10. Add the ``upstream`` repository:
+```bash
+git remote add upstream https://github.com/ansible-collections/community.mysql.git
+```
+
+11. Update your local ``main`` branch:
+```bash
+git fetch upstream
+git rebase upstream/main
+```
+
+12. Create a branch for your changes:
+```bash
+git branch -b name_of_my_branch
+```
+
+13. It can be a good approach is to start with writing integration tests if applicable.
+
+Note: If there are any difficulties with writing the tests or you are not sure if the case can be covered, feel free to skip this step.
+If needed, other contributors can help you with it later.
+
+All integration tests are stored in ``tests/integration/targets`` subdirectories.
+You are interested in a subdirectory containing a name of module you are going to change.
+For example, if you are fixing the ``mysql_user`` module, its tests are in ``tests/integration/targets/test_mysql_user/tasks``
+
+There is the ``main.yml`` file which includes other test files.
+Look for a suitable one to integrate your tests or create and include a dedicated one.
+You can use one of the existing as a draft.
+
+When fixing a bug, write a task which reproduces the bug from the issue.
+
+Put the reported case in the tests, then run integration tests with the following command:
+```bash
+ansible-test integration test_mysql_user --docker -vvv > ~/test.log
+```
+If the tests do not want to run, first, check you complete step 3 of this guide.
+
+If the tests ran successfully, there are usually two possible outcomes:
+a) If the bug has not appeared and the tests have passed successfully, ask the reporter to provide more details. The bug can be not a bug actually or can relate to a particular software version used or specifics of local environment configuration.
+
+b) The bug has appeared and the tests has failed as expected showing the reported symptoms.
+
+14. Fix the bug.
+
+15. Run ``flake8``:
+```bash
+flake8 /path/to/changed/file
+```
+It is worth installing and running and ``flake8`` against the changed file(s) first.
```

Comment:
> against the changed file(s) first.

Does the order of the steps need changing?

Reply:
@gundalow do you mean the extra `and` (Fixed, thanks!) or changing the step with sanity tests?
If the second one - I added the note. It's better to run flake8 first because sanity tests will show everything (except unused imports) but they run much longer. Running flake8 first will probably save time.

Reply:
At the moment you have
Fix bug 
Run flake8

Should the order be 
Run flake8 
Run tests

Reply:
Now it's:
```
119 14. Fix the bug.
121 15. Run ``flake8``:
128 16. Run sanity tests:
135 17. Run integration tests:
```
Which seems to be fine to me:) Sorry, I haven't slept very well today, not sure I fully understand